src/Libraries/Storage/Adapters/GoogleDrive/GoogleDriveFileSystemAdapter.php

```diff
@@ -0,0 +1,237 @@
+        } catch (Exception $e) {
+            return false;
+        }
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function append(string $filename, string $content)
+    {
+        $fileContent = $this->get($filename);
+
+        return $this->put($filename, $fileContent . $content);
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function rename(string $oldName, string $newName): bool
+    {
+        try {
+            $data = [
+                'name' => $newName
+            ];
+
+            $this->googleDriveApp->rpcRequest(GoogleDriveApp::FILE_METADATA_URL . '/' . $oldName, $data, 'PATCH');
+            return true;
+        } catch (Exception $e) {
+            return false;
+        }
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function copy(string $source, string $dest = 'root'): bool
+    {
+        try {
+            $data = [
+                'parents' => [$dest]
+            ];
+
+            $this->googleDriveApp->rpcRequest(GoogleDriveApp::FILE_METADATA_URL . '/' . $source . '/copy', $data);
+            return true;
+        } catch (Exception $e) {
+            return false;
+        }
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function exists(string $filename): bool
+    {
+        return $this->isFile($filename);
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function size(string $filename)
+    {
+        try {
+            $meta = (array)$this->googleDriveApp->getFileInfo($filename);
+            return $meta['size'];
+        } catch (Exception $e) {
+            return false;
+        }
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function lastModified(string $filename)
+    {
+        try {
+            $meta = (array)$this->googleDriveApp->getFileInfo($filename);
+            return !empty($meta['modifiedTime']) ? strtotime($meta['modifiedTime']) : false;
+        } catch (Exception $e) {
+            return false;
+        }
+
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function remove(string $filename): bool
+    {
+        try {
+            $this->googleDriveApp->rpcRequest(GoogleDriveApp::FILE_METADATA_URL . '/' . $filename, [], 'DELETE');
+            return true;
+        } catch (Exception $e) {
+            return false;
+        }
+
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function isFile(string $filename): bool
+    {
+        try {
+            $meta = (array)$this->googleDriveApp->getFileInfo($filename);
+
+            return $meta['kind'] === "drive#file" && $meta['mimeType'] != GoogleDriveApp::FOLDER_MIMETYPE;
+        } catch (Exception $e) {
+            return false;
+        }
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function isDirectory(string $dirname): bool
+    {
+        try {
+            $meta = (array)$this->googleDriveApp->getFileInfo($dirname);
+
+            return $meta['kind'] === "drive#file" && $meta['mimeType'] === GoogleDriveApp::FOLDER_MIMETYPE;
```

Comment:
What is the difference between file and folder?

Reply:
Looks like `isDirectory()` and `isFile()` cehcking the same thing

Reply:
Basically google drive folder is a file with its own mimetype (see the constant). isDirectory() checks if the file has that mimetype, on the other hand isFile() checks to see if the file doesn't have that mimetype.